docs/namespace-scoped-setup.md

```diff
@@ -0,0 +1,325 @@
+
+**Example: Configuring BMO to watch its own deployed namespace:**
+
+To configure the operator to automatically watch the same namespace it's
+deployed in, modify the `config/base/manager.yaml` file. Add the following
+environment variable configuration under spec.template.spec.containers.env`:
+
+```yaml
+# config/base/manager.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: controller-manager
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+      - name: manager
+        env:
+        - name: WATCH_NAMESPACE
+          value: foo,bar,quo
+```
+
+## 3\. Restricting RBAC Permissions (Roles and RoleBindings)
+
+When BMO operates in a namespace-scoped mode, its Kubernetes permissions must
+be updated from cluster-wide (`ClusterRole`, `ClusterRoleBinding`) to
+namespace-specific (`Role`, `RoleBinding`).
+
+**Steps:**
+
+1.**Generate manifests:**
+
+  ```bash
+    make manifests
+  ```
+
+2.**Add manifests for each namespaces:**
+
+  ```yaml
+  apiVersion: v1
+  kind: Namespace
+  metadata:
+    name: bar
+  ---
+  apiVersion: v1
+  kind: Namespace
+  metadata:
+    name: foo
+  ---
+  apiVersion: v11. **Change/Add Roles for each namespaces:**
+  kind: Namespace
+  metadata:
+    name: quo
+  ---
+  ```
+
+2.**Change/Add Roles for each namespaces:**
+
+  ```yaml
+  ---
+  apiVersion: rbac.authorization.k8s.io/v1
+  kind: Role
+  metadata:
+    name: baremetal-operator-manager-role
+    namespace: bar
+  rules:
+  - apiGroups:
+    - ""
+    resources:
+    - events
+    verbs:
+    - create
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups:
+    - ""
+    resources:
+    - secrets
+    verbs:
+    - delete
+    - get
+    - list
+    - update
+    - watch
+  - apiGroups:
+    - metal3.io
+    resources:
+    - baremetalhosts
+    - bmceventsubscriptions
+    - dataimages
+    - firmwareschemas
+    - hardwaredata
+    - hostfirmwarecomponents
+    - hostfirmwaresettings
+    - preprovisioningimages
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups:
+    - metal3.io
+    resources:
+    - baremetalhosts/finalizers
+    - dataimages/finalizers
+    - hardware/finalizers
+    - hostfirmwarecomponents/finalizers
+    verbs:
+    - update
+  - apiGroups:
+    - metal3.io
+    resources:
+    - baremetalhosts/status
+    - bmceventsubscriptions/status
+    - dataimages/status
+    - firmwareschemas/status
+    - hostfirmwarecomponents/status
+    - hostfirmwaresettings/status
+    - preprovisioningimages/status
+    verbs:
+    - get
+    - patch
+    - update
+  - apiGroups:
+    - metal3.io
+    resources:
+    - hostupdatepolicies
+    verbs:
+    - get
+    - list
+    - update
+    - watch
+  ```
+
+4.**Change/Add RoleBindings for each namespaces:**
+
+  ```yaml
+  ---
+  apiVersion: rbac.authorization.k8s.io/v1
+  kind: RoleBinding
+  metadata:
+    name: baremetal-operator-manager-rolebinding
+    namespace: bar
```

Comment:
Is it possible to skip the namespace for the Role and RoleBinding here and just add it through kustomize/kubectl? That was what I had in mind, i.e. we could provide these manifests and they can then easily be applied in any namespace that should be watched, like this:
```bash
kubectl -n foo -f bmo-namespaced-rbac.yaml
kubectl -n bar -f bmo-namespaced-rbac.yaml
kubectl -n baz -f bmo-namespaced-rbac.yaml
```

Reply:
Yes, it is possible. This would be a manual work. 
By doing this we will not use kustomize plugin.

Reply:
Good! The plugin is really useful I think, but it is nice to have an easy, manual way of doing it also. Some will prefer it that way for one reason or another :slightly_smiling_face: